api/src/test/java/uk/gov/hmcts/payment/api/mapper/CreditAccountDtoMapperTest.java

```diff
@@ -0,0 +1,153 @@
+        PaymentAllocation allocation1 = PaymentAllocation.paymentAllocationWith()
+            .receivingOffice("receiving-office")
+            .unidentifiedReason("reason")
+            .paymentAllocationStatus(PaymentAllocationStatus.paymentAllocationStatusWith().name("Transferred").build()).build();
+        paymentAllocations1.add(allocation1);
+        StatusHistory statusHistory = StatusHistory.statusHistoryWith()
+                                                .status("success")
+                                                .externalStatus("external-status")
+                                                .errorCode("")
+                                                .message("")
+                                                .dateCreated(new Date(2021,1,1))
+                                                .dateUpdated(new Date(2021,1,10))
+                                                .build();
+        List<StatusHistory> statusHistories = new ArrayList<StatusHistory>();
+        statusHistories.add(statusHistory);
+        payment1 = Payment.paymentWith()
+            .siteId("siteId")
+            .paymentChannel(PaymentChannel.paymentChannelWith().name("bulk scan").build())
+            .serviceType("service-type")
+            .caseReference("case-reference")
+            .reference("RC-1612-3710-5335-6484")
+            .ccdCaseNumber("ccd-case-number")
+            .bankedDate(new Date(2021,1,1))
+            .documentControlNumber("document-control-number")
+            .paymentMethod(PaymentMethod.paymentMethodWith().name("pay-method").build())
+            .amount(new BigDecimal("100.00"))
+            .paymentStatus(PaymentStatus.SUCCESS)
+            .dateCreated(new Date(2020,10,1))
+            .currency("GBP")
+            .statusHistories(statusHistories)
+            .id(1).paymentAllocation(paymentAllocations1).build();
+        payments.add(payment1);
+        PaymentFee fee = PaymentFee.feeWith().feeAmount(new BigDecimal("100.00")).ccdCaseNumber("ccd-case-number").calculatedAmount(new BigDecimal("100.00")).build();
+        List<PaymentFee> paymentFees = new ArrayList<PaymentFee>();
+        paymentFees.add(fee);
+        Remission remission = Remission.remissionWith()
+            .hwfAmount(new BigDecimal("10.00"))
+            .build();
+        List<Remission> remissionList = new ArrayList<Remission>();
+        remissionList.add(remission);
+        feeLink = PaymentFeeLink.paymentFeeLinkWith()
+            .paymentReference("payment-reference")
+            .fees(paymentFees)
+            .payments(payments)
+            .remissions(remissionList)
+            .build();
+        payment1.setPaymentLink(feeLink);
+
+    }
+
+    @Test
+    public void testToCreateCreditAccountPaymentResponse(){
+        PaymentDto paymentDto = creditAccountDtoMapper.toCreateCreditAccountPaymentResponse(feeLink);
+        assertEquals("RC-1612-3710-5335-6484",paymentDto.getReference());
```

Comment:
Instead of checking only value from mapper try to add few more assert statements for different fields